### ISSUE TITLE
🐛 Markdown parser infinite loop

### ISSUE THREAD

**Issue by dyc3** [biomejs/biome#11638]
### Environment information

<details>

```bash
playground

```

</details>

### What happened?

https://biomejs.dev/playground/?tab=formatter&pane=Diagnostics&language=md#code=LQAgAHgACgAgACAAYABgAGAACgAgACAAYABgAGAAIAAgAAoA

````markdown
- x
  ```
  ```  
````

See that it doesn't show the output. If you monitor your cpu usage, you should see one of your cores pinned at 100%.

Found by ecosystem CI

### Expected result

It should parse.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
